### PR TITLE
Pratima Active Checkbox Toggle Fix

### PR DESCRIPTION
**Pull request by PratimaSingh02**
# Description
image

## Related PRS (if any):
This frontend PR is related to the latest dev backend

## Main changes explained:
- Changed 'id' of Checkbox components in TaskTables.jsx
- Set 'htmlfor' attribute's value to 'id' prop in Checkbox component 

## How to test:
1. check into current branch
2. do `npm install` and `npm run start:local` to run this PR locally
3. Clear site data/cache
4. Log in as an Admin or Owner
5. Go to Reports -> Projects -> Any project
6. Click on “Active” text (Not the checkbox but the text itself)
7. Verify that clicking on "Active" text toggles the Active checkbox only.

## Screenshots or videos of changes:
image
image

### CODE REVIEW

**Review by meetpadhiar4 — DISMISSED** [review]
Hi, I tested your PR and it works as intended. Please refer the video attached.


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/67577022/0d60879f-5f23-44e3-9696-465bb19f38dd

File: src/components/Reports/TasksTable/TasksTable.jsx
Comment (Abi-Liu) [inline src/components/Reports/TasksTable/TasksTable.jsx:26]: Remove unnecessary useEffect.

Just as a tip, this useEffect is unnecessary even when testing because`isAssigned` is set in state. Whenever `isAssigned` changes, the component will rerender and all the code inside the file will rerun, including the `console.log(isAssigned)`

**Review by Abi-Liu — CHANGES_REQUESTED** [review]
Looks good, clicking on the label will now toggle the checkbox. Just a small bit of testing code that is still inside the pull request that needs to be removed.


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/59585815/97759bfc-19f1-450b-9824-4a79ccc2192b

**Review by lacnoskillz — DISMISSED** [review]
Hello @PratimaSingh02 
I have successfully pulled and tested your changes and they work as expected.
Toggling the active check box no longer toggles the Assign check box. It also works on Reports > People and Reports >Teams pages.
Nice find and great fix.
See below video

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/114305309/00b24467-ff7f-4f7e-8948-b4f405ee713c

**Review by SanketKaware — DISMISSED** [review]
LGTM.

I have checked the previous code - earlier it used to toggle the "Assign" checkbox instead of the "Active" checkbox when we clicked on the "Active" text itself. 

Now that issue is resolved.

Here is my video review for reference - 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/17793708/f1486b23-22cd-4cff-b7e3-b1837c4d5904

**Comment by dhairya-mehta** [comment]
Hey Pratima, I verified that clicking on "Active" text toggles the Active checkbox only, good job!
image

**Review by yaow62023 — DISMISSED** [review]
Hi, i have test your PR, when click the Active check box is only check the Active not Assigned. Check the video below for more details:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/137429365/9e14dc47-be26-4c84-bfc4-a8bf69046dbd

**Review by sameersdeshpande — DISMISSED** [review]
Hello @PratimaSingh02 
I checked this PR and this is working as expected. 
Approving and attaching the video here.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/87390339/324ee9de-bafe-4752-bfc4-948b134a22a1

**Review by mohabbasd — DISMISSED** [review]
Hello, I tested you PR and it works perfectly. Below is a video for reference.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/91215370/863bc8c9-a390-4d53-9e7f-23e5fac68320

File: src/components/Reports/TasksTable/TasksTable.jsx
Comment (PratimaSingh02) [reply src/components/Reports/TasksTable/TasksTable.jsx:26]: Good catch! I have removed it. Thank You

**Review by Abi-Liu — APPROVED** [review]
Looks good!

**Comment by one-community** [comment]
Thank you all, moving this to final review!

**Review by KurtisIvey — APPROVED** [review]
Great job, everything functions as intended and the overall goal of this fix was addressed. I also loved the fixes addressed in your second commit. No errors are present from any of the files that you worked on as well in the front or backend. Therefore, you have my approval on it. 


Screenshot 2024-04-05 at 4 28 01 PM

**Comment by one-community** [comment]
Thank you all, merging as soon as a bug blocking our backend merges to Main is solved.

**Pull request merged**
